source/components/publicodesUtils.tsx

```diff
@@ -24,31 +88,43 @@ export const FullName = ({ dottedName }) => (
 export const title = (rule) =>
 	rule.titre ||
 	capitalise0(splitName(rule.dottedName)[splitName(rule.dottedName).length - 1])
+
 // Publicodes's % unit is strangely handlded
 // the nodeValue is * 100 to account for the unit
 // hence we divide it by 100 and drop the unit
-export const correctValue = (evaluated) => {
+export function correctValue(evaluated: EvaluatedNode): number | undefined {
 	const { nodeValue, unit } = evaluated
 
+	if (nodeValue == undefined || typeof nodeValue !== 'number') {
+		return undefined
+	}
+
 	const result = unit?.numerators.includes('%') ? nodeValue / 100 : nodeValue
 	return result
 }
 
-const ruleSumNode = (rules, rule) => {
+function ruleSumNode(
+	rules: NGCRulesNodes,
+	rule: NGCRuleNode
+): string[] | undefined {
 	const formula = rule.rawNode.formule
 
-	return formula.somme?.map((name) =>
+	if (!formula || !formula['somme']) {
+		return undefined
+	}
+
+	return formula['somme'].map((name: string) =>
 		coreUtils.disambiguateReference(rules, rule.dottedName, name)
 	)
 }
 
 export const extractCategoriesNamespaces = (
-	rules,
-	engine,
-	parentRule = 'bilan'
+	rules: { [key: string]: { icônes: any; couleur: any } },
```

Comment:
accent circonflexe dans une variable 🤢

Reply:
C'est la norme dans NGC visiblement. C'est pour correspondre aux attributs publicodes.

Reply:
Bien noté !